pkg/util/database.go

```diff
@@ -0,0 +1,209 @@
+
+const (
+	MaxDbEntry = 1000
+	MaxKeySz   = 16
+	MaxValSz   = 32
+)
+
+// Init Initialise the database
+// Initialises the data map and creates json file
+func (db *JsonData) Init() error {
+	db.dataMap = make(map[string]string)
+	file, err := os.Create(JsonDB)
+	defer file.Close()
+	if err != nil {
+		return fmt.Errorf("creating json file failed: %s", err)
+	}
+
+	// initialise with empty data
+	jsonData, err := json.Marshal(db.dataMap)
+	if err != nil {
+		fmt.Print("json marshal failed")
+		return err
+
+	}
+	err = os.WriteFile(JsonDB, jsonData, 0655)
+
+	if err != nil {
+		fmt.Print("writing to file failed")
+		return err
+	}
+	return err
+}
+
+// Get function reads json file and
+// returns a map containing data
+func (db *JsonData) Get(key string) (val string, err error) {
+
+	db.m.RLock()
+	defer db.m.RUnlock()
+
+	data, err := os.ReadFile(JsonDB)
+	if err != nil {
+		fmt.Print("reading json file failed")
+		return "", err
+	}
+
+	// unmarshal json data to data map
+	err = json.Unmarshal([]byte(data), &db.dataMap)
+	if err != nil {
+		fmt.Print(" unmarshalling json failed")
+		return "", err
+	}
+	if _, ok := db.dataMap[key]; !ok {
+		return "", fmt.Errorf("key not present")
+	}
+
+	return db.dataMap[key], nil
+}
+
+// Update function
+// writes the data map into json file
+func (db *JsonData) Update(key, value string) error {
+	db.m.Lock()
+	defer db.m.Unlock()
+
+	data, err := os.ReadFile(JsonDB)
+	if err != nil {
+		fmt.Print("reading json file failed")
+		return err
+	}
+	err = json.Unmarshal([]byte(data), &db.dataMap)
+	if err != nil {
+		fmt.Print(" unmarshalling json data failed")
+		return err
+	}
+
+	if len(db.dataMap) > MaxDbEntry {
+		return fmt.Errorf("database maximum size %d reached", MaxDbEntry)
+	}
+	db.dataMap[key] = value
+
+	jsonData, err := json.Marshal(db.dataMap)
+	if err != nil {
+		return fmt.Errorf("could not marshal json: %s\n", err)
+
+	}
+	err = os.WriteFile(JsonDB, jsonData, 0655)
+
+	if err != nil {
+		fmt.Print(" writing to json file failed")
+		return err
+	}
+	return nil
+}
+
+func (db *JsonData) Delete(key string) error {
+
+	db.m.Lock()
+	defer db.m.Unlock()
+	data, err := os.ReadFile(JsonDB)
+	if err != nil {
+		fmt.Print("reading json file failed")
+		return err
+	}
+
+	// unmarshal json data to data map
+	err = json.Unmarshal([]byte(data), &db.dataMap)
+	if err != nil {
+		return fmt.Errorf("could not unmarshal json: %s\n", err)
+	}
+	// delete the entry
+	delete(db.dataMap, key)
+	jsonData, err := json.Marshal(db.dataMap)
+	if err != nil {
+		return fmt.Errorf("could not marshal json: %s\n", err)
+
+	}
+	err = os.WriteFile(JsonDB, jsonData, 0655)
+
+	if err != nil {
+		return err
+	}
+	return nil
+
+}
+
+// Contains function
+// Checks for the presence for entry in the database
+func (db *JsonData) Contains(key string) bool {
+	if _, err := db.Get(key); err != nil {
+		return false
+	}
+	return true
+}
+
+// MemData Map as a database to store key-value pairs
+type MemData JsonData
```

Comment:
- If the internal structure is a map, I think MemData should be the main structure name and JsonData its variant. 
- MemData should have a proper name like e.g. InMemoryDatabase. We're not using C so names could be more descriptive.
- Different types of database should be in different files (same with the interface: it needs to be in its own file)